specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -250,7 +250,7 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
         fieldValue: string | number | null,
         fieldName: string
       ): boolean => {
-        if (other.id !== null && other.id === this.resource.id) return false;
+        if (typeof Number(other.id) !== 'number' && other.id === this.resource.id) return false;
```

Comment:
hmm, could you double check `typeof Number(other.id) !== 'number'`?

I am pretty sure this is always true, regardless of value

```ts
Number('a')
> NaN
typeof Number('a')
> "number"

Reply:
"non a number" is a number after all 🤪

Reply:
also, I belive we have an eslint rule about not using the `Number` constructor - because usually a better alternative exists (be it `Number.parseInt/parseFloat`, `f.parseInt/parseFloat` or some other type of check)

don't forget that `Number.isNaN()` also exists

Reply:
Ah, thank you for the alternatives. 
Well, the problem was that `other.id` was undefined rather than null, and while this would be fixed by removing the strict inequality, there may have been odd behaviors with Javascript's implicit casting. 

Now that I look into it, this condition _is_ always true. I assumed Number(undefined) would not be a number, but it is NaN, as you said. (And even worse, Number(null) is 0).

Reply:
then either do:
`other.id !== null && other.id !== undefined`
or:
`other.id != null`
both are equivalent (2nd one is only ever going to do auto-casting for null and undefined)
or:
`(other.id ?? undefined) === undefined`
(paranthesis are important due to `??` having lower precedence than `===` 🤦‍♀️)

Reply:
I'd have to look into this more, but interestingly `other.id != null` and `(other.id ?? undefined) === undefined` had different behaviors. 

I think i'm okay settling for  `other.id != null` since it only casts for null/undefined values.

Reply:
Fixed in 91eb76b31086d66c876158ea2f533f0b360a40c8